File: challenges/2026-09-22-roll-call-scanner-glitch/solution.java
import java.util.*;

public class solution {

    public static List<Integer> findScanGlitch(List<Integer> scans) {
        // TODO: implement
        return List.of(0, 0);
    }

    public static void main(String[] args) {
        List<Object[]> tests = new ArrayList<>();
        tests.add(new Object[]{"example 1", List.of(1, 2, 2, 4), List.of(2, 3)});
        tests.add(new Object[]{"example 2", List.of(1, 1), List.of(1, 2)});
        tests.add(new Object[]{"example 3", List.of(3, 2, 3, 4, 5, 1), List.of(3, 6)});
        tests.add(new Object[]{"edge: duplicate is smallest value, missing is largest", List.of(2, 2), List.of(2, 1)});
        tests.add(new Object[]{"edge: larger roster", List.of(7, 3, 4, 5, 6, 3, 1), List.of(3, 2)});

        for (Object[] test : tests) {
            String name = (String) test[0];
            @SuppressWarnings("unchecked")
            List<Integer> scans = (List<Integer>) test[1];
            @SuppressWarnings("unchecked")
            List<Integer> expected = (List<Integer>) test[2];
            List<Integer> actual = findScanGlitch(scans);
            String status = actual.equals(expected) ? "PASS" : "FAIL";
            System.out.println("[" + status + "] " + name + ": expected=" + expected + " actual=" + actual);
        }
    }
}

File: challenges/2026-09-22-roll-call-scanner-glitch/solution.go
package main

import "fmt"

func findScanGlitch(scans []int) []int {
	// TODO: implement
	return []int{0, 0}
}

type testCase struct {
	name     string
	scans    []int
	expected []int
}

func main() {
	tests := []testCase{
		{"example 1", []int{1, 2, 2, 4}, []int{2, 3}},
		{"example 2", []int{1, 1}, []int{1, 2}},
		{"example 3", []int{3, 2, 3, 4, 5, 1}, []int{3, 6}},
		{"edge: duplicate is smallest value, missing is largest", []int{2, 2}, []int{2, 1}},
		{"edge: larger roster", []int{7, 3, 4, 5, 6, 3, 1}, []int{3, 2}},
	}

	for _, tc := range tests {
		got := findScanGlitch(tc.scans)
		status := "FAIL"
		if len(got) == len(tc.expected) && got[0] == tc.expected[0] && got[1] == tc.expected[1] {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%v actual=%v\n", status, tc.name, tc.expected, got)
	}
}
